### PR TITLE
Otherconsoleargs

### PR DESCRIPTION
**Pull request by acesiddhu**
## Description
Added Support to take in Other console Arguments though consoleparemters class. Added relevant unit tests as well

### CODE REVIEW

File: test/TranslationLayer.UnitTests/ConsoleParametersTests.cs [hunk 1]
```diff
@@ -26,5 +26,25 @@ public void LogFilePathShouldEnsureDoubleQuote()
 
             Assert.IsTrue(result.StartsWith("\""));
         }
+
+        [TestMethod]
+        public void OtherConsoleArgumentsShouldPersistValue()
```
Comment (singhsarab) [inline test/TranslationLayer.UnitTests/ConsoleParametersTests.cs]: What if I give invalid arguments ?

**Review by singhsarab — DISMISSED** [review]
Please mention the testing done.
I am just thinking, should we do some basic parsing to make sure arguments passed do not result in vstest.console crash. 
Minimally we should validate we get a proper error message.

File: test/TranslationLayer.UnitTests/ConsoleParametersTests.cs (hunk 1 above)
Comment (acesiddhu) [reply test/TranslationLayer.UnitTests/ConsoleParametersTests.cs]: Invalid aArguments are not processed and understood here. that job is done by vstest.console.exe

**Comment by acesiddhu** [comment]
testing is in progress. overall arg processing and validation needs to be done at vstest.console.exe layer IMO

**Comment by smadala** [comment]
@acesiddhu What are the other console parameters we want to use? the example in unit tests "/framework:4.5 /logger:trx" won't give desirable results.

Can you please add acceptance test?

File: src/vstest.console/CommandLine/CommandLineOptions.cs [hunk 2]
```diff
@@ -229,7 +229,16 @@ public bool ShouldCollectSourceInformation
             {
                 return IsDesignMode;
             }
-        }        
+        }
+
+        /// <summary>
+        /// The instance of Executor that is used in current instance of process.
+        /// </summary>
+        public Executor CurrentExecutor
```
Comment (singhsarab) [inline src/vstest.console/CommandLine/CommandLineOptions.cs]: Don't like this. Singleton, maybe?
Should we have some ProcessorManager, that we use in the executor and also pass on?

File: src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs [hunk 3]
```diff
@@ -10,6 +10,8 @@ namespace Microsoft.VisualStudio.TestPlatform.Client.DesignMode
     using System.Threading.Tasks;
 
     using Microsoft.VisualStudio.TestPlatform.Client.RequestHelper;
+    using Microsoft.VisualStudio.TestPlatform.Common;
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs]: >using [](start = 4, length = 5)

Remove.

**Comment by acesiddhu** [comment]
@dotnet-bot please test Windows_NT / Release Build

**Comment by acesiddhu** [comment]
@dotnet-bot please test Windows_NT / Debug Build

**Comment by acesiddhu** [comment]
Closing this PR. The design is not consistent. Ideally we should make RunSettingsManager as an service which takes care of Replacing Nodes and merging Runsetting when needed

**Pull request closed without merging**